tests/FaktorySpec.hs

```diff
@@ -44,3 +45,23 @@ spec = describe "Faktory" $ do
 
     jobs <- readMVar processedJobs
     jobs `shouldMatchList` ["a", "b", "HALT"]
+
+  it "doesn't crash on a consumer timeout" $ do
```

Comment:
Could you add some comments to the stages of this test, I'm not sure I'm totally understanding the intent of each distinct step.

Reply:
Will push the comment in a moment. As it's written, I'm not sure this test is doing what I intended. The one I'm about to push does, but requires an artificial delay of ~2 seconds, which always bothers me in test suites.

I'm curious if you think it's worth it. The other options are

1. No test
2. A pending test with a note to un-pend if you're touch the area it's covering

Reply:
For context on what's covered by this test: a null response from the server needs correctly handling through `recvUnsafe`, `commandJSON`, `fetchJob`, and `runWorker`. Any of these could mi-interpret the null as an error or misinterpret errors as the null. No existing test cover this path, and this PR was working on exactly it. Hence this test. :shrug: